bundles/org.openhab.binding.sncf/src/main/java/org/openhab/binding/sncf/internal/handler/StationHandler.java

```diff
@@ -0,0 +1,223 @@
+package org.openhab.binding.sncf.internal.handler;
+
+import static org.openhab.binding.sncf.internal.SncfBindingConstants.*;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.sncf.internal.SncfException;
+import org.openhab.binding.sncf.internal.dto.Passage;
+import org.openhab.core.i18n.LocationProvider;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link StationHandler} is responsible for handling commands, which are sent
+ * to one of the channels.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class StationHandler extends BaseThingHandler {
+    private static final DateTimeFormatter NAVITIA_DATE_FORMAT = DateTimeFormatter.ofPattern("yyyyMMdd'T'HHmmssZ");
+
+    private final Logger logger = LoggerFactory.getLogger(StationHandler.class);
+    private final LocationProvider locationProvider;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @NonNullByDefault({}) String stationId;
+    private @NonNullByDefault({}) String zoneOffset;
+
+    public StationHandler(Thing thing, LocationProvider locationProvider) {
+        super(thing);
+        this.locationProvider = locationProvider;
+    }
+
+    @Override
+    public void initialize() {
+        logger.trace("Initializing the Station handler for {}", getThing().getUID());
+
+        stationId = (String) getConfig().get("stopPointId");
+
+        if (thing.getProperties().isEmpty()) {
+            discoverAttributes(stationId);
+        }
+        zoneOffset = ZoneId.of(thing.getProperties().get(TIMEZONE)).getRules().getOffset(Instant.now()).getId()
+                .replace(":", "");
+
+        updateStatus(ThingStatus.ONLINE);
+        scheduleRefresh(ZonedDateTime.now().plusSeconds(2));
+    }
+
```

Comment:
As usual, you should override `bridgeStatusChanged` because the default behaviour is only to update the thing status but your code in initialize() will not be run again.

Reply:
Done

Reply:
Not exactly.
Like in `initialize`, you should calll `discoverAttributes` to get the timezone.
In fact, the easier for you would be to just call `initialize` in `bridgeStatusChanged `.

Reply:
Done.

Reply:
For me, the best implementation would have been to start from the super class code, something like that:
```
    @Override
    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE
                && getThing().getStatusInfo().getStatusDetail() == ThingStatusDetail.BRIDGE_OFFLINE) {
            initialize();
        } else if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
        }
    }
```

Reply:
Ok I think I understand. It is probably due to your task cancelation (freeRefreshJob).
By creating this additional, you avoid canceling yourself ;)
I am not sure this is a clean implementation but let's keep it like that.